### DISCUSSION TITLE
Make __pub_id available in custom runners called via the peer interface

### DISCUSSION THREAD

**Discussion by aplovich** [saltstack/salt#67497] [Feature Requests]
**Is your feature request related to a problem? Please describe.**
I'm looking to create a runner that signs x509 certs for minions.  The minions will use the peer runner interface to request certs from the master.  The flow for this involves calling an external API that does the signing, so the existing x509 state won't work because it requires access to a signing cert.

The master will render the minion's pillar to determine if its authorized to get the cert it requested.  This means the custom runner needs to know who called it.  So far I haven't found a way to get this information.

**Describe the solution you'd like**
Looking through the x509 module, I found that Salt populates **kwargs with information about who made the call:
https://github.com/saltstack/salt/blob/master/salt/modules/x509.py#L1047

It'd be nice if the same could be done inside a custom runner: add "**kwargs" to the runner function's signature, and Salt places __pub_id into it.

**Describe alternatives you've considered**
I've considered implementing this as an execution module,  however I'd explicitly like to run this on the master, so using a runner seems like a better way to call this out.

**Comment by welcome** [comment]
Hi there! Welcome to the Salt Community! Thank you for making your first contribution. We have a lengthy process for issues and PRs. Someone from the Core Team will follow up as soon as possible. In the meantime, here’s some information that may help as you continue your Salt journey.
Please be sure to review our [Code of Conduct](https://github.com/saltstack/salt/blob/master/CODE_OF_CONDUCT.md). Also, check out some of our community resources including:

  - [Community Wiki](https://github.com/saltstack/community/wiki)
  - [Salt’s Contributor Guide](https://docs.saltproject.io/en/master/topics/development/contributing.html)
  - [Join our Community Slack](https://join.slack.com/t/saltstackcommunity/shared_invite/zt-3av8jjyf-oBQ2M0vhXOhJpNpRkPWBvg)
  - [IRC on LiberaChat](https://web.libera.chat/#salt)
  - [Salt Project YouTube channel](https://www.youtube.com/channel/UCpveTIucFx9ljGelW63-BWg)
  - [Salt Project Twitch channel](https://www.twitch.tv/saltprojectoss)

There are lots of ways to get involved in our community. Every month, there are around a dozen opportunities to meet with other contributors and the Salt Core team and collaborate in real time. The best way to keep track is by subscribing to the Salt Community Events Calendar.
If you have additional questions, email us at saltproject@vmware.com. We’re glad you’ve joined our community and look forward to doing awesome things with you!

**Comment by lkubb** [comment]
For the record, a workaround for this issue can be found in the `vault` modules. They sign the request with the minion key and check it on the master side, explicitly found here:

https://github.com/saltstack/salt/blob/b7f9c2fd724a62df54ee598082fd51ddac32fd44/salt/utils/vault.py#L60-L66
https://github.com/saltstack/salt/blob/b7f9c2fd724a62df54ee598082fd51ddac32fd44/salt/runners/vault.py#L24-L52
https://github.com/saltstack/salt/blob/b7f9c2fd724a62df54ee598082fd51ddac32fd44/salt/runners/vault.py#L181-L198
